src/components/Button/Button-v2.tsx

```diff
@@ -0,0 +1,167 @@
+
+  /**
+   * The size of the button on screen
+   */
+  size?: Extract<Size, 'sm' | 'md' | 'lg'>;
+
+  /**
+   * The variant of the default tertiary button.
+   */
+  context?: 'default' | 'standalone';
+
+  /**
+   * Icon from the set of defined EDS icon set, when `iconLayout` is used.
+   */
+  icon?: IconName;
+
+  /**
+   * Allows configuation of the icon's positioning within `Button`.
+   *
+   * - When set to a value besides `"none"`, an icon must be specified.
+   * - When `"icon-only"`, `aria-label` must be given a value.
+   */
+  iconLayout?: 'none' | 'left' | 'right' | 'icon-only';
+
+  /**
+   * Status (color) variant for `Button`.
+   *
+   * **Default is `"default"`**.
+   */
+  variant?: 'default' | 'critical' | 'inverse';
+
+  /**
+   * Whether the width of the button is set to the full layout.
+   */
+  isFullWidth?: boolean;
+
+  /**
+   * Whether `Button` is set to disabled state (disables interaction and updates appearance).
+   */
+  isDisabled?: boolean;
+
+  /**
+   * Loading state passed down from higher level used to trigger loader and text change.
+   */
+  isLoading?: boolean;
+};
+
+/**
+ * `import {Button} from "@chanzuckerberg/eds";`
+ *
+ * Component for making buttons that do not navigate the user to another page. Use button to trigger actions, menus,
+ * or other in-page activity.
+ *
+ * - If you need to style a navigation anchor, please use the `Link` component.
+ * - If you need to style a different element or component to
+ *   look like a button or link, you can use the `ClickableStyle` component.
+ */
+export const Button = forwardRef<HTMLButtonElement, ButtonV2Props>(
+  (
+    {
+      children,
+      className,
+      context,
+      icon = 'empty-circle',
+      iconLayout = 'none',
+      isDisabled,
+      isFullWidth,
+      isLoading,
+      type = 'button',
+      rank = 'primary',
+      size = 'lg',
+      variant = 'default',
+      ...other
+    },
+    ref,
+  ) => {
+    const componentClassName = clsx(
+      styles['button'],
+      context && clsx(styles[`button--context-${context}`]),
+      iconLayout && clsx(styles[`button--layout-${iconLayout}`]),
+      isDisabled && clsx(styles['button--disabled']),
+      isFullWidth && clsx(styles['button--full-width']),
+      isLoading && clsx(styles['button--loading']),
+      rank && clsx(styles[`button--${rank}`]),
+      size && clsx(styles[`button--${size}`]),
+      variant && clsx(styles[`button--variant-${variant}`]),
+      className,
+    );
+
+    const buttonContentClassName = clsx(
+      styles['button__text'],
+      isLoading && styles['button--is-loading'],
+    );
+
+    const ariaLabel =
+      iconLayout === 'icon-only' && !other['aria-label']
+        ? children
+        : other['aria-label'];
+
+    return (
+      <button
+        aria-label={ariaLabel}
+        className={componentClassName}
+        disabled={isDisabled}
+        ref={ref}
+        tabIndex={isDisabled ? -1 : undefined}
```

Comment:
I think a button with `disabled` already isn't focusable. Maybe we don't need this?

Reply:
I think you're right... for some reason this was in the old button and I didn't scrutinize it. I will double-check